hw/misc/hermes.c

```diff
@@ -210,6 +221,111 @@ static void hermes_bar0_write(void *opaque, hwaddr addr, uint64_t val,
 {
 }
 
+static int __do_dma(struct HermesState *hermes, hwaddr desc_addr,
+                       uint8_t num_desc, bool h2c)
+{
+    struct hermes_dma_desc *desc;
+    hwaddr src_addr, dst_addr, nxt_addr;
+    uint8_t nxt_adj;
+    void *bar4_base = memory_region_get_ram_ptr(&hermes->bar4->mem_reg);
+
+    trace_hermes_dma_num_adj(num_desc - 1);
+    desc = malloc(num_desc * sizeof(*desc));
```

Comment:
In general use the glib helper functions for memory allocations.

Here, use `g_new(struct hermes_dma_desc, num_desc)`.

Reply:
Thanks, fixed